docs/registry.md

```diff
@@ -98,40 +105,51 @@ version = 2
   auth = ""
   identitytoken = ""
 ```
+
 The meaning of each field is the same with the corresponding field in `.docker/config.json`.
 
 Please note that auth config passed by CRI takes precedence over this config.
 The registry credential in this config will only be used when auth config is
 not specified by Kubernetes via CRI.
 
-After modify this config, you need restart the `containerd` service.
+After modifying this config, you need to restart the `containerd` service.
 
-### Configure Registry Credentials Example - GCR with _json_key Authentication
+### Configure Registry Credentials Example - GCR with service account key authentication
 
-Create a gcp account with gcr, do all the steps to enable receiving a
-pushed image for a gcr instance, including the generation and download of a
-new _json_key (for a new service account user.) To make sure your
-gcr registry is working with _json_key authentication let's login and
-push an image to your gcr instance: *This step is not necessary if you have
-already pushed an image to your gcr instance.*
+If you don't already have Google Container Registry (GCR) set-up then you need to do the following steps:
 
-```bash
-$ docker login -u _json_key -p "$(cat key.json)" gcr.io
-$ docker push gcr.io/your-gcr-instance-id/busybox
-$ docker logout gcr.io
-```
+* Create a Google Cloud Platform (GCP) account and project if not already created (see [GCP getting started](https://cloud.google.com/gcp/getting-started))
+* Enable GCR for your project (see [Quickstart for Container Registry](https://cloud.google.com/container-registry/docs/quickstart))
+* For authentication to GCR: Create [service account and JSON key](https://cloud.google.com/container-registry/docs/advanced-authentication#json-key)
+* The JSON key file needs to be downloaded to your system from the GCP console
+* For access to the GCR storage: Add service account to the GCR storage bucket with storage admin access rights (see [Granting permissions](https://cloud.google.com/container-registry/docs/access-control#grant-bucket))
+
+Refer to [Pushing and pulling images](https://cloud.google.com/container-registry/docs/pushing-and-pulling) for detailed information on the above steps.
 
-Generate a single line for the _json_key file that you downloaded:
+> Note: The JSON key file is a multi-line file and can be cumbersome to use its contents as a key outside of file. It is worthwhile generating a single line format output of the file. One way of doing this is using the `jq` tool as follows: `jq -c . key.json`
```

Comment:
s/and can/and it can/
s/its/the contents/
s/of file/of the file/

Reply:
Updated